mailpoet/lib/Subscribers/SubscribersEmailCountsController.php

```diff
@@ -0,0 +1,142 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Subscribers;
+
+use MailPoet\Entities\ScheduledTaskEntity;
+use MailPoet\Entities\ScheduledTaskSubscriberEntity;
+use MailPoet\Entities\SendingQueueEntity;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoetVendor\Carbon\Carbon;
+use MailPoetVendor\Doctrine\ORM\EntityManager;
+
+class SubscribersEmailCountsController {
+  private $processedTaskIdsTableCreated = false;
+
+  /** @var EntityManager */
+  private $entityManager;
+
+  /** @var string */
+  private $subscribersTable;
+
+  public function __construct(
+    EntityManager $entityManager
+  ) {
+    $this->entityManager = $entityManager;
+    $this->subscribersTable = $this->entityManager->getClassMetadata(SubscriberEntity::class)->getTableName();
+  }
+
+  public function updateSubscribersEmailCounts(?\DateTimeInterface $dateLastProcessed, int $batchSize, ?int $startId = null): array {
+    $sendingQueuesTable = $this->entityManager->getClassMetadata(SendingQueueEntity::class)->getTableName();
+    $scheduledTasksTable = $this->entityManager->getClassMetadata(ScheduledTaskEntity::class)->getTableName();
+    $scheduledTaskSubscribersTable = $this->entityManager->getClassMetadata(ScheduledTaskSubscriberEntity::class)->getTableName();
+
+    $connection = $this->entityManager->getConnection();
+
+    $dayAgo = new Carbon();
+    $dayAgoIso = $dayAgo->subDay()->toDateTimeString();
+
+    $startId = (int)$startId;
+
+    [$countSubscribersToUpdate, $endId] = $this->countAndMaxOfSubscribersInRange($startId, $batchSize);
+    if (!$countSubscribersToUpdate) {
+      return [0, 0];
+    }
+
+    // Temporary table with processed tasks from threshold date up to yesterday
+    $processedTaskIdsTable = 'processed_task_ids';
+    if (!$this->processedTaskIdsTableCreated) {
+      $queryParams = [];
+      $processedTaskIdsTableSql = "
+        CREATE TEMPORARY TABLE IF NOT EXISTS {$processedTaskIdsTable}
+        (INDEX task_id_ids (id))
+        SELECT DISTINCT task_id as id FROM {$sendingQueuesTable} as sq
+          JOIN {$scheduledTasksTable} as st ON sq.task_id = st.id
+          WHERE st.processed_at IS NOT NULL
+          AND st.processed_at < :dayAgo";
+      $queryParams['dayAgo'] = $dayAgoIso;
+
+      if ($dateLastProcessed) {
+        $processedTaskIdsTableSql .= " AND st.processed_at >= :dateFrom";
+        $carbonDateLastProcessed = Carbon::createFromTimestamp($dateLastProcessed->getTimestamp());
+        $dateFromIso = ($carbonDateLastProcessed->subDay())->toDateTimeString();
+        $queryParams['dateFrom'] = $dateFromIso;
+      }
+
+      $resultQuery = $connection->executeQuery($processedTaskIdsTableSql, $queryParams);
+      $this->processedTaskIdsTableCreated = true;
+
+      if ($resultQuery->rowCount() === 0) return [0,0];
+    }
+
+    // Temporary table needed for UPDATE query
+    // mySQL does not allow to modify the same table used in the select
```

Comment:
On my local env, I have MariaDB and I was able to UPDATE with one query, but it doesn't work with mySQL:
```
MailPoet\Subscribers\SubscribersEmailCountsControllerTest::testItCalculatesTotalSubscribersEmailCounts MailPoetVendor\Doctrine\DBAL\Exception\DriverException: An exception occurred while executing ' UPDATE mp_mailpoet_subscribers as s SET s.emails_count = + IFNULL( ( SELECT count(task_ids.id) from mp_mailpoet_subscribers s2 JOIN mp_mailpoet_scheduled_task_subscribers sts ON s2.id = sts.subscriber_id JOIN processed_task_ids task_ids ON task_ids.id = sts.task_id WHERE s2.id = s.id GROUP BY s.id ) ,0) WHERE s.id >= ? AND s.id <= ? ' with params [0, "3"]: SQLSTATE[HY000]: General error: 1093 You can't specify target table 's' for update in FROM clause
```